### PR TITLE
added tests for generated value sequence table

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
added tests for generated ID table.

also changed mapping of numeric/decimal to string. However this requires JDBC support

### CODE REVIEW

File: src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java [hunk 1]
```diff
@@ -89,10 +89,18 @@ public SpannerTableExporter(SpannerDialect spannerDialect) {
             + " " + col.getSqlType(this.spannerDialect, metadata)
             + (col.isNullable() ? this.spannerDialect.getNullColumnString() : " not null")));
 
-    return new String[]{
-        MessageFormat.format(this.createTableTemplate, table.getQuotedName(),
+    ArrayList<String> statements = new ArrayList<>();
+    statements.add(MessageFormat.format(this.createTableTemplate, table.getQuotedName(),
             colsAndTypes.toString(),
-            primaryKeyColNames)};
+        primaryKeyColNames));
```
Comment (dzou) [inline src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java]: nit: indentation of `primaryKeyColNames` does not seem quite right.

File: src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java [hunk 2]
```diff
@@ -90,8 +90,8 @@ public SpannerDialect() {
     registerColumnType(Types.NCLOB, "STRING(MAX)");
     registerColumnType(Types.BLOB, "BYTES(MAX)");
 
-    registerColumnType(Types.DECIMAL, "FLOAT64");
-    registerColumnType(Types.NUMERIC, "FLOAT64");
+    registerColumnType(Types.DECIMAL, "STRING(MAX)");
```
Comment (dzou) [inline src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java]: Do you intend to include these changes with the sequence stuff?

File: src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java (hunk 1 above)
Comment (ChengyuanZhao) [reply src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java]: Fixed.

File: src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java (hunk 2 above)
Comment (ChengyuanZhao) [reply src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java]: reverted. I guess we should wait until we have a driver that supports these two types before we make changes

**Pull request merged**